app/src/main/java/org/sopt/and/presentation/login/LogInViewModel.kt

```diff
@@ -0,0 +1,86 @@
+package org.sopt.and.presentation.login
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import org.sopt.and.R
+import org.sopt.and.data.dataremote.model.request.RequestSignInDto
+import org.sopt.and.domain.entity.UserLogInInfo
+import org.sopt.and.domain.entity.UserToken
+import org.sopt.and.domain.usecase.DataStoreSetTokenUseCase
+import org.sopt.and.domain.usecase.SignInUserUseCase
+import javax.inject.Inject
+
+@HiltViewModel
+class LogInViewModel @Inject constructor(
+    private val signInUserUseCase: SignInUserUseCase,
+    private val dataStoreSetTokenUseCase: DataStoreSetTokenUseCase
+) : ViewModel() {
+    private val _loginState = MutableStateFlow(UserLogInInfo())
+    val loginState = _loginState.asStateFlow()
+
+    private val _signInSideEffect = MutableSharedFlow<SignInSideEffect>()
+    val signInSideEffect get() = _signInSideEffect.asSharedFlow()
+
+
+    fun setUserName(userName: String) {
+        _loginState.update {
+            it.copy(
+                userName = userName
+            )
+        }
+    }
+
+    fun setPassword(password: String) {
+        _loginState.update {
+            it.copy(
+                password = password
+            )
+        }
+    }
+
+    private suspend fun signInUser(request: RequestSignInDto): Result<UserToken> =
+        signInUserUseCase(request)
+
+    private suspend fun setToken(token: String) {
+        dataStoreSetTokenUseCase.invoke(token)
```

Comment:
invoke는 이름없이도 호출이 가능합니다!
```suggestion
        dataStoreSetTokenUseCase(token)
```

Reply:
invoke가 무엇인지 공부해보시면 좋을 듯 합니다 !

Reply:
invoke는 코틀린에서 연산자라고 불리는 이름없이 호출가능한 함수군요.. 메모해두겠습니다 감사합니다!!